core/src/test/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java

```diff
@@ -254,7 +255,7 @@ private IndexShard newShard(boolean primary, DiscoveryNode node, IndexMetaData i
         public int indexDocs(final int numOfDoc) throws Exception {
             for (int doc = 0; doc < numOfDoc; doc++) {
                 final IndexRequest indexRequest = new IndexRequest(index.getName(), "type", Integer.toString(docId.incrementAndGet()))
-                    .source("{}");
+                    .source("{}").waitForActiveShards(ActiveShardCount.NONE);
```

Comment:
I remember now why I had to put this in place.... the `IndexingOp` which extends `ReplicationOperation` passes a cluster state `Supplier` that always gives `null`... and `ReplicationOperation#checkActiveShardCount` depends on a non-null cluster state. In this case, the better solution is to override `checkActiveShardCount` in `IndexingOp`, which is what I've done and will push shortly.